src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1135,16 +1166,26 @@ public FormValidation doCheckSshKeysCredentialsId(@AncestorInPath ItemGroup cont
                 // Don't do anything if the user is only reading the configuration
                 return FormValidation.ok();
             }
-            if (value == null || value.isEmpty()){
-                return FormValidation.error("No ssh credentials selected");
-            }
 
-            SSHUserPrivateKey sshCredential = getSshCredential(value, context);
-            String privateKey = "";
-            if (sshCredential != null) {
-                privateKey = sshCredential.getPrivateKey();
+            String privateKey;
+
+            if (System.getProperty(SSH_PRIVATE_KEY_FILEPATH, "").isEmpty()) {
+                if (value == null || value.isEmpty()) {
+                    return FormValidation.error("No ssh credentials selected");
+                }
+
+                SSHUserPrivateKey sshCredential = getSshCredential(value, context);
+                if (sshCredential != null) {
+                    privateKey = sshCredential.getPrivateKey();
+                } else {
+                    return FormValidation.error("Failed to find credential \"" + value + "\" in store.");
+                }
             } else {
-                return FormValidation.error("Failed to find credential \"" + value + "\" in store.");
+                EC2PrivateKey k = fetchPrivateKeyFromDisk();
+                if (k == null) {
+                    return FormValidation.error("Failed to find private key file " + System.getProperty(SSH_PRIVATE_KEY_FILEPATH));
+                }
+                privateKey = k.getPrivateKey();
```

Comment:
Let's suppose a user is setting the System property AND selecting a credentials, which is even valid as it was in use previously. This message might make the user get confused as the credentials selected is valid.

This whole method now should add a warning as well everytime that the credentials will be overridden by the system property. `FormValidation.aggregate` will do it